src/integration/java/io/pinecone/integration/dataPlane/UpsertAndQueryServerlessTest.java

```diff
@@ -0,0 +1,236 @@
+import io.pinecone.helpers.RandomStringBuilder;
+import io.pinecone.proto.DescribeIndexStatsResponse;
+import io.pinecone.unsigned_indices_model.QueryResponseWithUnsignedIndices;
+import io.pinecone.unsigned_indices_model.ScoredVectorWithUnsignedIndices;
+import io.pinecone.unsigned_indices_model.VectorWithUnsignedIndices;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.openapitools.client.model.IndexModelSpec;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import static io.pinecone.commons.IndexInterface.buildUpsertVectorWithUnsignedIndices;
+import static io.pinecone.helpers.AssertRetry.assertWithRetry;
+import static io.pinecone.helpers.BuildUpsertRequest.*;
+import static io.pinecone.helpers.IndexManager.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class UpsertAndQueryServerlessTest {
+    private static Index index;
+    private static AsyncIndex asyncIndex;
+    private static final int dimension = 3;
+    private static final Struct emptyFilterStruct = Struct.newBuilder().build();
+
+    @BeforeAll
+    public static void setUp() throws InterruptedException {
+        String apiKey = System.getenv("PINECONE_API_KEY");
+        String indexType = IndexModelSpec.SERIALIZED_NAME_SERVERLESS;
+        Pinecone pinecone = new Pinecone(apiKey);
+
+        String indexName = findIndexWithDimensionAndType(pinecone, dimension, indexType);
+        if (indexName.isEmpty()) indexName = createNewIndex(pinecone, dimension, indexType);
+        index = pinecone.createIndexConnection(indexName);
+        asyncIndex = pinecone.createAsyncIndexConnection(indexName);
+    }
+
+    @AfterAll
+    public static void cleanUp() {
+        index.close();
+        asyncIndex.close();
+    }
+
+    @Test
+    public void upsertOptionalVectorsAndQueryIndexSyncTest() throws Exception {
+        int numOfVectors = 5;
+        DescribeIndexStatsResponse describeIndexStatsResponse1 = index.describeIndexStats(emptyFilterStruct);
+        // Confirm the starting state by verifying the dimension of the index
+        assertEquals(describeIndexStatsResponse1.getDimension(), dimension);
+
+        // upsert vectors with required + optional parameters
+        List<String> upsertIds = getIdsList(numOfVectors);
+        int topK = 5;
+        String namespace = RandomStringBuilder.build("ns", 8);
+        List<Float> values = generateVectorValuesByDimension(dimension);
+        List<Long> sparseIndices = generateSparseIndicesByDimension(dimension);
+        List<Float> sparseValues = generateVectorValuesByDimension(dimension);
+        Struct metadataStruct = generateMetadataStruct();
+
+        List<VectorWithUnsignedIndices> vectors = new ArrayList<VectorWithUnsignedIndices>(numOfVectors);
+
+        for (String id : upsertIds) {
```

Comment:
Updated example for async upsert(List<Vector>, String namespace) and querying to confirm all params are upserted.
Will be adding extensive suite of tests in upcoming PR.